src/Mod/Part/App/GeometryPyImp.cpp

```diff
@@ -217,7 +217,8 @@ Py::Boolean GeometryPy::getConstruction(void) const
 
 void  GeometryPy::setConstruction(Py::Boolean arg)
 {
-    getGeometryPtr()->Construction = arg;
+    if(getGeometryPtr()->getClassTypeId() != Part::GeomPoint::getClassTypeId())

```

Comment:
It is under epigraph: a) Modify the api so that no unintended use is made.

If I am not wrong (I think I remember it from an old discussion), there is no point in making a point of type construction. If I am wrong and generally in FreeCAD it is useful to make a point construction=true. Then this shall *not* be merged AND the sketcher code will work under the other epigraph: b) "document so that developers are aware of the consequences of setting a point of type construction".

This BSpline deliverable introduces a fourth type of geometry type of sketcher geometry (other than normal, construction, external), the "construction point" (making use of the perceived idea that a construction point did not make sense in FreeCAD). A construction point is programmability created to indicate the solver that such point has fixed parameters and that it shall not try to move it. The idea behind this is that very complex geometry uses OCC functions for handling this geometry outside the solver.

I do not have strong feelings about this commit, if you decide towards a) we merged if you decide for b) I document better the issue in the code. Just let me know.